packages/firestore/src/core/component_provider.ts

```diff
@@ -114,73 +120,69 @@ export class IndexedDbComponentProvider implements ComponentProvider {
           initialUser
         )
       : new MemorySharedClientState();
-    components.sharedClientState.onlineStateHandler = onlineState =>
-      components.syncEngine!.applyOnlineStateChange(
+    this.sharedClientState.onlineStateHandler = onlineState =>
+      this.syncEngine!.applyOnlineStateChange(
```

Comment:
`this.syncEngine` is already declared as force unwrapped. Is the `!` necessary?

Reply:
Fixed